dataclients/kubernetes/routegroup.go

```diff
@@ -513,49 +518,120 @@ func transformRouteGroup(ctx *routeGroupContext) ([]*eskip.Route, error) {
 	return explicitGroupRoutes(ctx)
 }
 
+func splitHosts(hosts []string, domains []string) ([]string, []string) {
+	internalHosts := []string{}
+	externalHosts := []string{}
+
+	for _, host := range hosts {
+		for _, d := range domains {
+			if strings.HasSuffix(host, d) {
+				internalHosts = append(internalHosts, host)
+			} else {
+				externalHosts = append(externalHosts, host)
+			}
+		}
+	}
+
+	return internalHosts, externalHosts
+}
+
 func (r *routeGroups) convert(s *clusterState, df defaultFilters) ([]*eskip.Route, error) {
 	var rs []*eskip.Route
 
-	hostRoutes := make(map[string][]*eskip.Route)
 	for _, rg := range s.routeGroups {
+		var internalHosts []string
+		var externalHosts []string
+
 		hosts := rg.Spec.UniqueHosts()
-		ctx := &routeGroupContext{
-			clusterState:          s,
-			defaultFilters:        df,
-			routeGroup:            rg,
-			hosts:                 hosts,
-			hostRx:                createHostRx(hosts...),
-			hostRoutes:            make(map[string][]*eskip.Route),
-			hasEastWestHost:       hasEastWestHost(r.options.KubernetesEastWestDomain, hosts),
-			eastWestEnabled:       r.options.KubernetesEnableEastWest,
-			eastWestDomain:        r.options.KubernetesEastWestDomain,
-			provideHTTPSRedirect:  r.options.ProvideHTTPSRedirect,
-			httpsRedirectCode:     r.options.HTTPSRedirectCode,
-			backendsByName:        mapBackends(rg.Spec.Backends),
-			backendNameTracingTag: r.options.BackendNameTracingTag,
+		if len(r.options.KubernetesEastWestRangeDomains) == 0 {
+			externalHosts = hosts
+		} else {
+			internalHosts, externalHosts = splitHosts(hosts, r.options.KubernetesEastWestRangeDomains)
 		}
 
-		ri, err := transformRouteGroup(ctx)
-		if err != nil {
-			log.Errorf(
-				"[routegroup] error transforming %s/%s: %v.",
-				namespaceString(rg.Metadata.Namespace),
-				rg.Metadata.Name,
-				err,
-			)
+		backends := mapBackends(rg.Spec.Backends)
+
+		// If there's no host at all, or if there's any external hosts
+		// create it.
+		if len(externalHosts) != 0 || len(hosts) == 0 {
+			ctx := &routeGroupContext{
+				clusterState:          s,
+				defaultFilters:        df,
+				routeGroup:            rg,
+				hosts:                 externalHosts,
+				hostRx:                createHostRx(externalHosts...),
+				hostRoutes:            make(map[string][]*eskip.Route),
+				hasEastWestHost:       hasEastWestHost(r.options.KubernetesEastWestDomain, externalHosts),
```

Comment:
I think this `hasEastWestHost` uses deprecated option `KubernetesEastWestDomain` 
https://github.com/zalando/skipper/blob/62f282f0f25a7aa50635e2b1f6956c6000cfa35b/dataclients/kubernetes/kube.go#L152-L154
and maybe not needed here at all  (i.e. always false) as hosts were split (`splitHosts`) before and we know `externalHosts` do not contains east-west hosts...

Reply:
As far as i understand we just deprecated but didn't remove it yet.

Reply:
Also, I deprecated it after merging this PR, so we might want to revisit it soon.